packages/anoncreds/src/utils/credential.ts

```diff
@@ -49,6 +51,11 @@ export function encodeCredentialValue(value: unknown) {
     return value.toString()
   }
 
+  // If value is a number string with leading zero and not a decimal return as number string
+  if (isString(value) && !isNaN(Number(value)) && !isDecimal(value) && hasLeadingZero(value)) {
+    return value.toString()
+  }
+
```

Comment:
If the value is not a number we still need to encode it as a number.

So i would expand the next if statement to also add && !hadLeadingZero.

Otherwise it will reach the end as be encoded as any other string (by hashing)

Reply:
You either need to encode it as a number and remove the leading zero, or you need to encode it as a normal string

Reply:
I think adding `!hasLeadingZero(value)` in the next if statement resolves it.